pkg/generate/generate.go

```diff
@@ -47,18 +47,22 @@ func (c *Controller) applyGenerate(resource unstructured.Unstructured, gr kyvern
 	policy, err := c.pLister.Get(gr.Spec.Policy)
 	if err != nil {
 		if apierrors.IsNotFound(err) {
-			for _, e := range policy.Spec.Rules {
-				if e.Generation.Synchronize {
-					if err := c.client.DeleteResource(e.Generation.Kind, e.Generation.Namespace, e.Generation.Name, false); err != nil {
-						logger.V(4).Info("Generated resource is deleted")
-						return nil, err
-					}
+			resource, err := c.client.GetResource(gr.Spec.Resource.Kind, gr.Spec.Resource.Namespace, gr.Spec.Resource.Name)
```

Comment:
I think we can use Unstructured resource, rather than getting it from the cluster. More than that, we are returning on error when failed to get resource.